UI/src/view/linear_asset/laneModellingLayer.js

```diff
@@ -321,6 +332,18 @@
       return laneUtils.offsetByLaneNumber(linearAsset);
     };
 
+    var extractMiddleLinksOfChains = function (linearAssetChains, linksToConsider) {
+      return _.flatMap(linearAssetChains, function (chain) {
+        var  links = _.isEmpty(linksToConsider) ? chain : _.intersectionWith(chain, linksToConsider, _.isEqual);
+        return _.flatMap(_.groupBy(links, 'roadPartNumber'), function (chainByRoadPartNumber) {
+          var minAddressMValue = _.minBy(chainByRoadPartNumber, 'startAddrMValue');
+          var maxAddressMValue = _.maxBy(chainByRoadPartNumber, 'endAddrMValue');
+          var middleMValue = (maxAddressMValue.endAddrMValue - minAddressMValue.startAddrMValue)/2 + minAddressMValue.startAddrMValue;
+          return _.filter(chainByRoadPartNumber, function(linearAsset) { return linearAsset.startAddrMValue <= middleMValue && linearAsset.endAddrMValue > middleMValue; });
+        });
```

Comment:
Shouldn't be linearAsset.endAddrMValue >= middleMValue too ?

Reply:
No, the endAddrMValue will be checked as the startAddrMValue in the next link. Here we only want to draw the label in one link if we also checked for linearAsset.endAddrMValue >= middleMValue in the limit cases where the middle is equal to start/end we would return two links and draw two times for a chain.

Reply:
OK. understood